### PR TITLE
Notion : Pages Indexing , Cleanup , Recursive page crawling

### PR DESCRIPTION
**Pull request by Vivek-Lahole**
1. Notion Cleanup
2. API token strategy for Notion
3. Recursive Notion Page Calling
4. Error Logging

### CODE REVIEW

File: packages/apps/notion/src/services/notion.ts [hunk 1]
```diff
@@ -54,232 +70,318 @@ export default class NotionService extends TransactionBaseService {
...
         last_sync: new Date(),
       });
     } catch (error) {
-      console.error(error);
+      this.logger_.error(
+        `fetchNotionData : Error fetching Notion pages for ${org.id} organisation: ${error}`
+      );
     }
 
     this.logger_.info(
```
Comment (louismurerwa) [inline packages/apps/notion/src/services/notion.ts:99]: Remove  

```    this.logger_.info(
      `Finished oculation of Notion for ${org.id} organisation`
    );```

File: packages/apps/notion/src/services/notion.ts [hunk 2]
```diff
@@ -35,11 +51,11 @@ export default class NotionService extends TransactionBaseService {
...
-    return Readable.from(this.getNotionPagesAndBlocks(org));
+  async notionIndexableDocuments(org: Organisation) {
+    return Readable.from(this.fetchNotionData(org));
   }
 
-  async *getNotionPagesAndBlocks(
+  async *fetchNotionData(
     org: Organisation
   ): AsyncGenerator<IndexableDocument[]> {
     this.logger_.info(
```
[the remark this answers is not in the document]
Comment (Vivek-Lahole) [reply packages/apps/notion/src/services/notion.ts:61]: Also the Occulation complete statement as well ?

File: packages/apps/notion/src/strategies/notion.ts [hunk 3]
```diff
@@ -1,32 +1,55 @@
...
+    const stream = await this.notionService_.notionIndexableDocuments(org);
     stream.on("data", (documents) => {
       this.queueService_.sendBatch(APPS_INDEXING_TOPIC, documents);
     });
+
+    // Log The End of The Indexing Process
+    stream.on("end", () => {
+      this.logger_.success(
+        oculationNotionPagesActivity,
+        `processJob:Starting oculation of Notion Pages for ${org.id} organisation`
```
Comment (louismurerwa) [inline packages/apps/notion/src/strategies/notion.ts:43]: this.logger_.success(
        oculationNotionPagesActivity,
        `processJob:Finished oculation of Notion Pages for ${org.id} organisation`

File: packages/apps/notion/src/strategies/notion.ts [hunk 4]
```diff
@@ -1,32 +1,55 @@
...
+      );
+    });
+
+    // Log The Error of The Indexing Process
     stream.on("end", () => {
```
Comment (louismurerwa) [inline packages/apps/notion/src/strategies/notion.ts:48]: stream.on("error", () => {
      this.logger_.error(
        oculationNotionPagesActivity,
        `processJob:Error oculation of Notion Pages for ${org.id} organisation`
      );
    });

**Review by louismurerwa — APPROVED** [review]

**Pull request merged**